### PR TITLE
feat(azure): add new policies for Azure Synapse

### PR DESCRIPTION
**Pull request by taviassaf**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    We use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the other types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Each prefix should be accompanied by a scope that specifies the targeted framework. If uncertain, use 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sast|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

*Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.*

Fixes # (issue)

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes

### CODE REVIEW

**Review by lirshindalman — APPROVED** [review]
nice!!

File: checkov/terraform/checks/graph_checks/azure/SynapseLogMonitoringEnabledForSQLPool.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+metadata: 
+  id: "CKV2_AZURE_54"
+  name: "Ensure log monitoring is enabled for Synapse SQL Pool"
+  category: "GENERAL_SECURITY"
```
Comment (SteveVaknin) [inline checkov/terraform/checks/graph_checks/azure/SynapseLogMonitoringEnabledForSQLPool.yaml:4]: perhaps `LOGGING` category is a better fit?

File: checkov/terraform/checks/graph_checks/azure/SynapseLogMonitoringEnabledForSQLPool.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        - cond_type: attribute
+          resource_types:
+            - azurerm_synapse_sql_pool_extended_auditing_policy
+          attribute: log_monitoring_enabled
+          operator: exists
+          
+        - cond_type: attribute
+          resource_types:
+            - azurerm_synapse_sql_pool_extended_auditing_policy
+          attribute: 'log_monitoring_enabled'
```
Comment (SteveVaknin) [inline checkov/terraform/checks/graph_checks/azure/SynapseLogMonitoringEnabledForSQLPool.yaml:31]: redundant quotes  
```suggestion
          attribute: log_monitoring_enabled
```

File: checkov/terraform/checks/graph_checks/azure/SynapseWorkspaceHasExtendedAuditLogs.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+metadata: 
+  id: "CKV2_AZURE_53"
+  name: "Ensure Azure Synapse Workspace has extended audit logs"
+  category: "GENERAL_SECURITY"
```
Comment (SteveVaknin) [inline checkov/terraform/checks/graph_checks/azure/SynapseWorkspaceHasExtendedAuditLogs.yaml:4]: perhaps LOGGING category is a better fit?

File: checkov/terraform/checks/resource/azure/SynapseWorkspaceAdministratorLoginPasswordHidden.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+from checkov.terraform.checks.resource.base_resource_check import BaseResourceCheck
+from typing import List
+
+
+class SynapseWorkspaceAdministratorLoginPasswordHidden(BaseResourceCheck):
+    def __init__(self):
+        name = "Ensure Azure Synapse Workspace administrator login password is not exposed"
+        id = "CKV_AZURE_240"
+        supported_resources = ['azurerm_synapse_workspace']
+        categories = [CheckCategories.ENCRYPTION]
```
Comment (SteveVaknin) [inline checkov/terraform/checks/resource/azure/SynapseWorkspaceAdministratorLoginPasswordHidden.py:11]: perhaps SECRETS category is a better fit?

**Review by SteveVaknin — COMMENTED** [review]
Great work! 
Left some comments

**Review by SteveVaknin — APPROVED** [review]
@taviassaf Amazing first contribution! 👏 🔝

**Pull request merged**